src/plugins/intel_cpu/tests/functional/single_layer_tests/non_max_suppression.cpp

```diff
@@ -129,41 +130,49 @@ class NmsLayerCPUTest : public testing::WithParamInterface<NmsParams>, virtual p
         std::tie(bounds, targetInDims) = inShapeParams;
 
         if (!bounds.empty()) {
-            inputDynamicShapes = std::vector<ngraph::PartialShape>{{bounds[BATCHES], bounds[BOXES], 4}, {bounds[BATCHES], bounds[CLASSES], bounds[BOXES]}};
+            inputDynamicShapes = std::vector<ov::PartialShape>{{bounds[BATCHES], bounds[BOXES], 4}, {bounds[BATCHES], bounds[CLASSES], bounds[BOXES]}};
         } else {
             size_t batches, boxes, classes;
             std::tie(batches, boxes, classes) = targetInDims.front();
             ov::Dimension numBatches(batches), numBoxes(boxes), numClasses(classes);
-            inputDynamicShapes = std::vector<ngraph::PartialShape>{{numBatches, numBoxes, 4}, {numBatches, numClasses, numBoxes}};
+            inputDynamicShapes = std::vector<ov::PartialShape>{{numBatches, numBoxes, 4}, {numBatches, numClasses, numBoxes}};
         }
 
         for (const auto &ts : targetInDims) {
             size_t numBatches, numBoxes, numClasses;
             std::tie(numBatches, numBoxes, numClasses) = ts;
-            targetStaticShapes.push_back(std::vector<ngraph::Shape>{{numBatches, numBoxes, 4}, {numBatches, numClasses, numBoxes}});
+            targetStaticShapes.push_back(std::vector<ov::Shape>{{numBatches, numBoxes, 4}, {numBatches, numClasses, numBoxes}});
             if (maxOutBoxesType == ngraph::helpers::InputLayerType::PARAMETER) {
-                targetStaticShapes.back().push_back(ngraph::Shape{1});
+                targetStaticShapes.back().push_back(ov::Shape{1});
             }
         }
 
-        std::shared_ptr<ngraph::Node> maxOutBoxesPerClassNode;
+        std::shared_ptr<ov::Node> maxOutBoxesPerClassNode;
         auto params = ngraph::builder::makeDynamicParams(paramsPrec, inputDynamicShapes);
         params[0]->set_friendly_name("param_1");
         params[1]->set_friendly_name("param_2");
 
         if (maxOutBoxesType == ngraph::helpers::InputLayerType::PARAMETER) {
-            inputDynamicShapes.push_back(ngraph::PartialShape{1});
-            params.push_back(std::make_shared<ngraph::opset1::Parameter>(element::Type_t::i32, inputDynamicShapes.back()));
+            inputDynamicShapes.push_back(ov::PartialShape{1});
+            if (maxBoxPrec == ElementType::i64) {
+                params.push_back(std::make_shared<ov::opset1::Parameter>(element::Type_t::i64, inputDynamicShapes.back()));
+            } else {
+                params.push_back(std::make_shared<ov::opset1::Parameter>(element::Type_t::i32, inputDynamicShapes.back()));
+            }
```

Comment:
Can we just use `maxBoxPrec` as a parameter?
```suggestion
            params.push_back(std::make_shared<ov::opset1::Parameter>(maxBoxPrec, inputDynamicShapes.back()));
```